app/services/step_code/compliance/propose_step/base.rb

```diff
@@ -9,26 +9,34 @@ def initialize(checklist:, min_required_step:)
 
   def call
     self.final_step_reached = false
-    self.step = min_required_step
+    self.step = min_required_step || min_step
 
     until final_step_reached || step > max_step
       if requirements_met?
         self.step += 1
       else
-        self.step = nil if step == min_required_step
+        self.step = nil unless min_requirement_met?
         self.final_step_reached = true
       end
     end
-    self.step = step - 1 if step && step > min_required_step
+    self.step = step - 1 if min_requirement_met?
     return self
   end
 
   private
 
+  def min_requirement_met?
+    step && step > (min_required_step || min_step)
+  end
+
   def requirements_met?
     checkers.all? { |c| self.send(c).requirements_met? }
   end
 
+  def min_step
+    raise NotImplementedError, "#{__method__} is not implemented"
+  end
```

Comment:
not implemented on purpose I assume

Reply:
yes this is expected to be implemented in the child class